### PR TITLE
Fix Clang -Wshorten-64-to-32 warnings

### PR DESCRIPTION
**Pull request by wantehchang**
Finish the job of https://github.com/AOMediaCodec/libavif/pull/1037.

### CODE REVIEW

File: contrib/gdk-pixbuf/loader.c [hunk 1]
```diff
@@ -365,12 +365,11 @@ static gboolean avif_image_saver(FILE          *f,
...
 
                 if (quality < 0 || quality > 100) {
-
                     g_set_error(error,
                                 GDK_PIXBUF_ERROR,
                                 GDK_PIXBUF_ERROR_BAD_OPTION,
-                                "AVIF quality must be a value between 0 and 100; value \"%d\" is not allowed.",
-                                (int)quality);
+                                "AVIF quality must be a value between 0 and 100; value \"%ld\" is not allowed.",
+                                quality);
```
Comment (wantehchang) [inline contrib/gdk-pixbuf/loader.c:372]: `quality` is declared as `long` at line 335.

```
    long quality = 52; /* default; must be between 0 and 100 */
```

File: contrib/gdk-pixbuf/loader.c [hunk 2]
```diff
@@ -423,7 +422,7 @@ static gboolean avif_image_saver(FILE          *f,
         }
     }
 
-    max_quantizer = AVIF_QUANTIZER_WORST_QUALITY * ( 100 - CLAMP(quality, 0, 100)) / 100;
+    max_quantizer = AVIF_QUANTIZER_WORST_QUALITY * (100 - (int)quality) / 100;
```
Comment (wantehchang) [inline contrib/gdk-pixbuf/loader.c:425]: The `CLAMP(quality, 0, 100)` call is not necessary because of the check at line 367.

**Review by wantehchang — COMMENTED** [review]
@novomesk Please review contrib/gdk-pixbuf/loader.c. Thanks.

File: src/read.c [hunk 3]
```diff
@@ -3101,7 +3101,11 @@ static avifResult avifDecoderPrepareSample(avifDecoder * decoder, avifDecodeSamp
 
             avifDecoderItem * item = avifMetaFindItem(decoder->data->meta, sample->itemID);
             avifROData itemContents;
-            avifResult readResult = avifDecoderItemRead(item, decoder->io, &itemContents, sample->offset, bytesToRead, &decoder->diag);
+            if (sample->offset > SIZE_MAX) {
+                return AVIF_RESULT_BMFF_PARSE_FAILED;
+            }
+            size_t offset = (size_t)sample->offset;
```
Comment (jzern) [inline src/read.c:3107]: this could be const, but maybe that isn't too common in this code base. I haven't looked around.

**Review by jzern — APPROVED** [review]

**Pull request merged**